helm/idol-library/templates/aciserver/_deployment.tpl

```diff
@@ -53,12 +53,21 @@ spec:
         volumeMounts:
         - name: config-map
           mountPath: /etc/config/idol
+        {{- if $component.oauthToolConfigMap }}
+        - name: oauth-tool-cm
+          mountPath: /etc/config/oauth
+          readOnly: true
+        {{- end }}
```

Comment:
I think I would prefer if we start adding extra 'arguments' in the template (e.g. volumeMounts, volumes, env) and add to the list here. e.g. this would become:
```
{{ $volumeMounts := get . "volumeMounts" | default list }}
...
        {{ range $volumeMounts}}
        - {{ . | toYaml | nindent 10 }}
        {{- end }}
```